draft-ietf-mls-protocol.md

```diff
@@ -2301,9 +2371,12 @@ The client verifies the validity of a LeafNode using the following steps:
   * If the LeafNode appears in the `leaf_node` value of the UpdatePath in
     a Commit, verify that `leaf_node_source` is set to `commit`.
 
-* Verify that the following fields are unique among the members of the group:
-    * `signature_key`
-    * `encryption_key`
+* Verify that the leaf node's `signature_key` value does not appear in any other
+  leaf node in the group's ratchet tree.
+
+* Verify that the leaf node's `encryption_key` value does not appear in any
+  other node of the group's ratchet tree (including both leaf nodes and parent
+  nodes).
```

Comment:
What exactly does it mean to "appear in any other node"? Is it OK e.g. if a signature or an HPKE key of a leaf appears in an intermediate certificate in the X509 chain in the credential of another leaf? Should we also check that the keys don't appear in other places e.g. in the external signers extension or as the HPKE key for external joiners?